python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -1392,35 +1392,14 @@ def __init__(
         super().__init__(dtype)
         self.op = op
         self.children = (left, right)
-        if (
-            op in (plc.binaryop.BinaryOperator.ADD, plc.binaryop.BinaryOperator.SUB)
-            and plc.traits.is_chrono(left.dtype)
-            and plc.traits.is_chrono(right.dtype)
-            and not dtypes.have_compatible_resolution(left.dtype.id(), right.dtype.id())
-        ):
-            raise NotImplementedError("Casting rules for timelike types")
-        if op in (
-            plc.binaryop.BinaryOperator.MUL,
-            plc.binaryop.BinaryOperator.DIV,
-            plc.binaryop.BinaryOperator.TRUE_DIV,
-            plc.binaryop.BinaryOperator.FLOOR_DIV,
+        if not plc.binaryop.is_supported_operation(
+            self.dtype, left.dtype, right.dtype, op
         ):
-            if (
-                left.dtype.id() in dtypes.TIMELIKE_TYPES
-                and right.dtype.id() in dtypes.FLOATING_TYPES
-            ) or (
-                right.dtype.id() in dtypes.TIMELIKE_TYPES
-                and left.dtype.id() in dtypes.FLOATING_TYPES
-            ):
-                raise NotImplementedError(
-                    "No multiplying or dividing durations by floats"
-                )
-            if (
-                left.dtype.id() in dtypes.TIMELIKE_TYPES
-                and right.dtype.id() in dtypes.TIMELIKE_TYPES
-                and dtype.id() in dtypes.FLOATING_TYPES
-            ):
-                raise NotImplementedError("No dividing durations by durations")
+            raise NotImplementedError(
+                f"Operation {op.name} not supported "
+                f"for types {left.dtype.id().name} and {right.dtype.id().name} "
+                f"with output type {self.dtype}"
```

Comment:
This ones needs an .id().name as well, I think

Reply:
```suggestion
                f"with output type {self.dtype.id().name}"
```